Doc/faq/programming.rst

```diff
@@ -1741,11 +1741,32 @@ but effective way to define class private variables.  Any identifier of the form
 is textually replaced with ``_classname__spam``, where ``classname`` is the
 current class name with any leading underscores stripped.
 
+The identifier can be used unchanged within the class, but to access it outside
+the class, the mangled name must be used:
+
+.. code-block:: python
+
+   class A:
+       def __one(self):
+           return 1
+       def two(self):
+           return 2 * self.__one()
+
+   class B(A):
+       def three(self):
+           return 3 * self._A__one()
+
+   four = 4 * A()._A__one()
+
 This doesn't guarantee privacy: an outside user can still deliberately access
 the "_classname__spam" attribute, and private values are visible in the object's
```

Comment:
Isn't this a bit redundant now?

Reply:
Mmmh... that's right. How about the following paragraph:

```rst
The identifier can be used unchanged within the class, but to access it outside
the class, the mangled name must be used:

<EXAMPLE>

In particular, this does not guarantee privacy since an outside user can still
deliberately access the mangled attribute, and private values remain visible in
the object's ``__dict__``.  Many Python programmers never bother to use private
variable names at all.
```

**OR**

```diff
-In particular, this does not guarantee privacy since an outside user can still
-deliberately access the mangled attribute, and private values remain visible in
-the object's ``__dict__``.  Many Python programmers never bother to use private
-variable names at all.
+In particular, this does not guarantee privacy since an outside user can still
+deliberately access the mangled attribute; many Python programmers never bother
+to use private variable names at all.
```

@JelleZijlstra

Reply:
sounds good!

Reply:
First or second suggestion? (I think the second is easier to digest for a beginner)

Reply:
I like the second as it is more concise.